drivers/rtc/rtc_smartbond.c

```diff
@@ -0,0 +1,635 @@
+	struct rtc_smartbond_data *data = (struct rtc_smartbond_data *)dev->data;
+
+	k_mutex_lock(&data->lock, K_FOREVER);
+
+	data->update_cb = callback;
+	data->update_user_data = user_data;
+
+	if (data->update_cb) {
+		/* Enable asserting the RTC interrupt line when the second counter rolls over. */
+		RTC->RTC_INTERRUPT_ENABLE_REG = RTC_RTC_INTERRUPT_ENABLE_REG_RTC_SEC_INT_EN_Msk;
+	} else {
+		RTC->RTC_INTERRUPT_DISABLE_REG = RTC_RTC_INTERRUPT_DISABLE_REG_RTC_SEC_INT_DIS_Msk;
+	}
+
+	k_mutex_unlock(&data->lock);
+
+	return 0;
+}
+#endif
+
+struct rtc_driver_api rtc_smartbond_driver_api = {
+	.get_time = rtc_smartbond_get_time,
+	.set_time = rtc_smartbond_set_time,
+#if defined(CONFIG_RTC_ALARM)
+	.alarm_get_time = rtc_smartbond_alarm_get_time,
+	.alarm_set_time = rtc_smartbond_alarm_set_time,
+	.alarm_is_pending = rtc_smartbond_alarm_is_pending,
+	.alarm_set_callback = rtc_smartbond_alarm_set_callback,
+	.alarm_get_supported_fields = rtc_smartbond_alarm_get_supported_fields,
+#endif
+#if defined(CONFIG_RTC_UPDATE)
+	.update_set_callback = rtc_smartbond_update_set_callback,
+#endif
+#if defined(CONFIG_RTC_CALIBRATION)
+	.set_calibration = NULL,
+	.get_calibration = NULL,
+#endif
+};
+
+static void rtc_smartbond_100HZ_clock_cfg(void)
+{
+#define RTC_DIV_DENOM_1000  0
+#define RTC_DIV_DENOM_1024  1
+
+	const struct device * const dev = DEVICE_DT_GET(DT_NODELABEL(osc));
+	uint32_t lp_clk_rate;
+	uint32_t clk_rtcdiv_reg;
+
+	if (!device_is_ready(dev)) {
+		__ASSERT_MSG_INFO("Clock device is not ready");
+	}
+
+	if (clock_control_get_rate(dev, (clock_control_subsys_t)SMARTBOND_CLK_LP_CLK,
+								&lp_clk_rate) < 0) {
+		__ASSERT_MSG_INFO("Cannot extract LP clock rate");
+	}
+
+	clk_rtcdiv_reg = CRG_TOP->CLK_RTCDIV_REG;
+	CLK_RTCDIV_REG_SET_FIELD(DENOM, clk_rtcdiv_reg, RTC_DIV_DENOM_1000);
+	CLK_RTCDIV_REG_SET_FIELD(INT, clk_rtcdiv_reg, lp_clk_rate / 100);
+	CLK_RTCDIV_REG_SET_FIELD(FRAC, clk_rtcdiv_reg, (lp_clk_rate % 100) * 10);
+	CRG_TOP->CLK_RTCDIV_REG = clk_rtcdiv_reg;
+}
+
+static int rtc_smartbond_init(const struct device *dev)
+{
+	struct rtc_smartbond_data *data = (struct rtc_smartbond_data *)dev->data;
+
+	memset(data, 0, sizeof(*data));
+
+	/* Wakeup device from RTC events (alarm/roll over) */
+#if CONFIG_PM
+	bool is_xtal32m_enabled = !!DT_NODE_HAS_STATUS(DT_NODELABEL(xtal32m), okay);
+	int pdc_idx = da1469x_pdc_add(MCU_PDC_TRIGGER_RTC_ALARM, MCU_PDC_MASTER_M33,
+					is_xtal32m_enabled ? MCU_PDC_EN_XTAL : 0);
+
+	__ASSERT(pdc_idx >= 0, "Failed to add RTC PDC entry");
+	da1469x_pdc_set(pdc_idx);
+	da1469x_pdc_ack(pdc_idx);
+#endif
+
+	rtc_smartbond_100HZ_clock_cfg();
+
+	/* Timer and calendar counters will not reset after SW reset */
+	RTC->RTC_KEEP_RTC_REG |= RTC_RTC_KEEP_RTC_REG_RTC_KEEP_Msk;
+
+#if defined(CONFIG_RTC_ALARM) || defined(CONFIG_RTC_UPDATE)
+	IRQ_CONNECT(IRQN, IRQ_PRIO, smartbond_rtc_isr, DEVICE_DT_INST_GET(0), 0);
+	NVIC_ClearPendingIRQ(IRQN);
+	NVIC_EnableIRQ(IRQN);
+#endif
+
+	return 0;
+}
+
+
+#define RTC_INIT(inst) \
```

Comment:
You should namespace this macro as `RTC_` can run into naming conflicts with the rtc.h header :) something like `SMARTBOND_RTC_INIT`

Reply:
Fixed